File: src/nl/rubensten/texifyidea/action/group/SumatraActionGroup.java
package nl.rubensten.texifyidea.action.group;

import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.DataContext;
import com.intellij.openapi.actionSystem.DefaultActionGroup;
import nl.rubensten.texifyidea.TexifyIcons;
import nl.rubensten.texifyidea.run.SumatraConversationKt;

/**
 * @author Ruben Schellekens, Sten Wessel
 */
public class SumatraActionGroup extends DefaultActionGroup {

    @Override
    public boolean canBePerformed(DataContext context) {
        return SumatraConversationKt.isSumatraAvailable();
    }

    @Override
    public boolean hideIfNoVisibleChildren() {
        return true;
    }

    @Override
    public void update(AnActionEvent event) {
        super.update(event);
        event.getPresentation().setIcon(TexifyIcons.SUMATRA);
    }
}

File: src/nl/rubensten/texifyidea/action/group/SumatraActionGroup.kt
package nl.rubensten.texifyidea.action.group

import com.intellij.openapi.actionSystem.DataContext
import com.intellij.openapi.actionSystem.DefaultActionGroup
import nl.rubensten.texifyidea.run.sumatra.isSumatraAvailable

/**
 * @author Ruben Schellekens, Sten Wessel
 */
class SumatraActionGroup : DefaultActionGroup() {

    override fun canBePerformed(context: DataContext) = isSumatraAvailable

    override fun hideIfNoVisibleChildren() = true
}